.generated/Add.ts

```diff
@@ -0,0 +1,436 @@
+    JSON.parse(`{
+  "version": "v1.7.0",
+  "name": "Add",
+  "bytecode": "02020d40360100020402041600160100010200000202021605160016015f06160016015fa00016002a16012aa100a000160016010e0dce00010002",
+  "codeHash": "52d5893e97ce6b8d67d90fe1c51735e6e4f9946de732926fd160a0b50774f61b",
+  "fieldsSig": {
+    "names": [
+      "sub",
+      "result"
+    ],
+    "types": [
+      "ByteVec",
+      "U256"
+    ],
+    "isMutable": [
+      false,
+      true
+    ]
+  },
+  "eventsSig": [
+    {
+      "name": "Add",
+      "fieldNames": [
+        "x",
+        "y"
+      ],
+      "fieldTypes": [
+        "U256",
+        "U256"
+      ]
+    },
+    {
+      "name": "Add1",
+      "fieldNames": [
+        "a",
+        "b"
+      ],
+      "fieldTypes": [
+        "U256",
+        "U256"
+      ]
+    }
+  ],
+  "functions": [
+    {
+      "name": "add",
+      "usePreapprovedAssets": false,
+      "useAssetsInContract": false,
+      "isPublic": true,
+      "paramNames": [
+        "array"
+      ],
+      "paramTypes": [
+        "[U256;2]"
+      ],
+      "paramIsMutable": [
+        false
+      ],
+      "returnTypes": [
+        "[U256;2]"
+      ]
+    },
+    {
+      "name": "addPrivate",
+      "usePreapprovedAssets": false,
+      "useAssetsInContract": false,
+      "isPublic": false,
+      "paramNames": [
+        "array"
+      ],
+      "paramTypes": [
+        "[U256;2]"
+      ],
+      "paramIsMutable": [
+        false
+      ],
+      "returnTypes": [
+        "[U256;2]"
+      ]
+    }
+  ]
+}`)
+  );
+
+  readonly address: Address;
+  readonly contractId: string;
+  readonly groupIndex: number;
+  deployResult: SignDeployContractTxResult | undefined;
+
+  private constructor(
+    address: Address,
+    contractId: string,
+    groupIndex: number,
+    deployResult?: SignDeployContractTxResult
+  ) {
+    this.address = address;
+    this.contractId = contractId;
+    this.groupIndex = groupIndex;
+    this.deployResult = deployResult;
+  }
+
+  static async deploy(
+    signer: SignerProvider,
+    sub: string,
+    result: bigint,
+    _extraParams?: {
+      initialAttoAlphAmount?: bigint;
+      initialTokenAmounts?: Token[];
+      issueTokenAmount?: bigint;
+      gasAmount?: number;
+      gasPrice?: bigint;
+    }
+  ): Promise<Add> {
+    const _deployResult = await Add.contract.deploy(signer, {
+      initialFields: { sub: sub, result: result },
+      initialAttoAlphAmount: _extraParams?.initialAttoAlphAmount,
+      initialTokenAmounts: _extraParams?.initialTokenAmounts,
+      issueTokenAmount: _extraParams?.issueTokenAmount,
+      gasAmount: _extraParams?.gasAmount,
+      gasPrice: _extraParams?.gasPrice,
+    });
+    return new Add(
+      _deployResult.contractAddress,
+      _deployResult.contractId,
+      _deployResult.fromGroup,
+      _deployResult
+    );
+  }
+
+  static connect(
+    address: string,
+    deployResult?: SignDeployContractTxResult
+  ): Add {
```

Comment:
1. deployResult can be removed from the class
2. TypedChain uses `attach`, which sounds better
```
static attach(
    address: string,
  ): Add {
```